core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.ts

```diff
@@ -146,4 +151,44 @@ export class NgbdModalWorkflowExecutionsComponent implements OnInit {
         );
       });
   }
+
+  /* Pagination handler */
+  onPageIndexChange(pageIndex: number): void {
+    this.currentPageIndex = pageIndex;
+    this.changePaginatedExecutions();
```

Comment:
this function call is very expensive. Why is it necessary?

Reply:
Call the change function after page index changed in order to re-render table list with new items related to the page index.